<launch>
  <arg name="lidar_ip1" default="10.10.10.121" />
  <arg name="lidar_ip2" default="10.10.10.122" />

  <node pkg="tf" type="static_transform_publisher" name="scanner1_to_map" args="0 0 0 0 0 0 map laser_frame1 100" />

  <node pkg="tf" type="static_transform_publisher" name="scanner2_to_map" args="0 0 2 0 0 0 map laser_frame2 100" />

  <node name="PavoScanNode1"   pkg="pavo2_ros"  type="pavo2_scan_node"  respawn="false" output="screen">
  <remap from="/PavoScanNode1/scan1" to="/scan1" />
    <param name="frame_id"     type="string"   value="laser_frame1"/><!--配置frame_id-->
    <param name="scan_topic"   type="string"   value="scan1" /><!--设置激光雷达的topic名称--> 
    <param name="angle_min"    type="double"   value="-140.00" /><!--设置最小角度，单位度.取值范围[-140,140].default=-140.0-->
    <param name="angle_max"    type="double"   value="140.00"/><!--设置最大角度，单位度.取值范围[-140,140].default=140.0-->
    <param name="range_min"    type="double"   value="0.10" /><!--设置最小距离，单位米，default=0.10-->
    <param name="range_max"    type="double"   value="50.0" /><!--设置最大距离，单位米，default=20-->
    <param name="angle_resolution"  type="int"      value="32"/>
    <param name="motor_speed"  type="int"      value="50" /><!--default=50Hz-->
    <param name="angle_start"  type="int"      value="4000" /><!--设置起始角度，default=4000，即40度-->
    <param name="angle_end"  type="int"      value="32000" /><!--设置结束角度，default=32000，即320度-->
    <param name="intensity"     type="bool"     value="true"/><!--强度使能，使能强度true，不使能强度false，default=true -->
    <param name="inverted"     type="bool"     value="false"/><!--配置雷达是否倒装，倒装true，正装false-->
    <param name="echo_mode"     type="bool"     value="true"/><!--二回波使能，使能二回波true，不使能二回波false，default=true -->
    
    <param name="lidar_ip"     type="string"   value="$(arg lidar_ip1)" /><!--所要连接的雷达IP地址，即通过PavoView设置的雷达IP地址-->
    <param name="lidar_port"   type="int"      value="2368" /><!--雷达ip默认为10.10.10.121，端口号默认为2368-->
    
    <!--param name="host_ip"     type="string"   value="10.10.10.100" /--><!--所要绑定的网卡IP地址，即通过PavoView设置的雷达目标上位机地址-->
    <!--param name="host_port"   type="int"      value="2368" /--><!--所要绑定的端口-->
  </node>

 <node name="PavoScanNode2"   pkg="pavo2_ros"  type="pavo2_scan_node"  respawn="false" output="screen">
  <remap from="/PavoScanNode2/scan2" to="/scan2" />
    <param name="frame_id"     type="string"   value="laser_frame2"/><!--配置frame_id-->
    <param name="scan_topic"   type="string"   value="scan2" /><!--设置激光雷达的topic名称--> 
    <param name="angle_min"    type="double"   value="-140.00" /><!--设置最小角度，单位度.取值范围[-140,140].default=-140.0-->
    <param name="angle_max"    type="double"   value="140.00"/><!--设置最大角度，单位度.取值范围[-140,140].default=140.0-->
    <param name="range_min"    type="double"   value="0.10" /><!--设置最小距离，单位米，default=0.10-->
    <param name="range_max"    type="double"   value="50.0" /><!--设置最大距离，单位米，default=20-->
    <param name="angle_resolution"  type="int"      value="32"/>
    <param name="motor_speed"  type="int"      value="50" /><!--default=50Hz-->
    <param name="angle_start"  type="int"      value="4000" /><!--设置起始角度，default=4000，即40度-->
    <param name="angle_end"  type="int"      value="32000" /><!--设置结束角度，default=32000，即320度-->
    <param name="intensity"     type="bool"     value="true"/><!--强度使能，使能强度true，不使能强度false，default=true -->
    <param name="inverted"     type="bool"     value="false"/><!--配置雷达是否倒装，倒装true，正装false-->
    <param name="echo_mode"     type="bool"     value="true"/><!--二回波使能，使能二回波true，不使能二回波false，default=true -->
    
    <param name="lidar_ip"     type="string"   value="$(arg lidar_ip2)" /><!--所要连接的雷达IP地址，即通过PavoView设置的雷达IP地址-->
    <param name="lidar_port"   type="int"      value="2368" /><!--雷达ip默认为10.10.10.121，端口号默认为2368-->
   
  </node>    
  
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find pavo2_ros)/rviz/pavo2_scan_twin.rviz" />
</launch>


